### PR TITLE
executor: avoid unnecessary log format in slow query log

### PR DESCRIPTION
**Pull request by jackysp**
Due to perf cpu profile, avoid unnecessary log format in slow query log. PTAL @coocood

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -360,6 +360,14 @@ func (a *ExecStmt) buildExecutor(ctx context.Context) (Executor, error) {
 func (a *ExecStmt) logSlowQuery(txnTS uint64, succ bool) {
 	cfg := config.GetGlobalConfig()
 	costTime := time.Since(a.startTime)
+	threshold := time.Duration(cfg.Log.SlowThreshold) * time.Millisecond
+	level := log.GetLevel()
+	if costTime < threshold && level < log.DebugLevel {
```
Comment (zimulala) [inline executor/adapter.go]: I think using `costTime < threshold` is enough.

**Comment by zimulala** [comment]
Rest LGTM

File: executor/adapter.go (hunk 1 above)
Comment (jackysp) [reply executor/adapter.go]: If only costTime < threshold here, when level is Debug, the query log will miss.

File: executor/adapter.go [hunk 2]
```diff
@@ -360,6 +360,14 @@ func (a *ExecStmt) buildExecutor(ctx context.Context) (Executor, error) {
...
+		return
+	}
+	if costTime >= threshold && level < log.WarnLevel {
```
Comment (coocood) [inline executor/adapter.go]: The first condition can be removed.

Comment (jackysp) [reply executor/adapter.go]: Done. PTAL @coocood

**Comment by coocood** [comment]
LGTM

File: executor/adapter.go [hunk 3]
```diff
@@ -358,8 +358,16 @@ func (a *ExecStmt) buildExecutor(ctx context.Context) (Executor, error) {
 }
 
 func (a *ExecStmt) logSlowQuery(txnTS uint64, succ bool) {
+	level := log.GetLevel()
+	if level < log.WarnLevel {
```
Comment (winkyao) [inline executor/adapter.go]: level < log.WarnLevel ? log.ErrorLevel is less than log.WarnLevel, so why return here?

File: executor/adapter.go [hunk 4]
```diff
@@ -358,8 +358,16 @@ func (a *ExecStmt) buildExecutor(ctx context.Context) (Executor, error) {
...
+		return
+	}
 	cfg := config.GetGlobalConfig()
 	costTime := time.Since(a.startTime)
+	threshold := time.Duration(cfg.Log.SlowThreshold) * time.Millisecond
+	if costTime < threshold && level < log.DebugLevel {
```
Comment (winkyao) [inline executor/adapter.go]: ditto

File: executor/adapter.go (hunk 3 above)
Comment (jackysp) [reply executor/adapter.go]: There is no need to print slow query log If log level is Error or Fatal.

File: executor/adapter.go (hunk 4 above)
Comment (jackysp) [reply executor/adapter.go]: If cost time is less than threshold, and log level is not debug, do not print query log. This modification keeps the same logic as before.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**